compiler/desugared/name_resolution.mli

```diff
@@ -149,6 +150,10 @@ val get_scope : context -> IdentName.t Marked.pos -> ScopeName.t
 (** Find a scope definition from the typedefs, failing if there is none or it
     has a different kind *)
 
+val process_base_typ : context -> Surface.Ast.base_typ Marked.pos -> typ
+(** Convert a surface base type to an AST type *)
+(* Note: should probably be moved to a different module *)
```

Comment:
No since it's used a lot in `Name_resolution`, on which `From_surface` depends. Maybe it would make sense directly in `Ast` ?

The reason I asked is that this was internal to `Name_resolution`, but I now needed to expose the function to `From_surface` that uses it when translating toplevel defs and didn't want to duplicate the code. It's not that bad though, maybe we can wait and see if further evolutions warrant a new module for these conversions ?

Reply:
Yes I agree.